### ISSUE TITLE
Improve test suite

### ISSUE THREAD

**Issue by willclarktech** [confio/ts-relayer#89]
This issue is for gathering potential improvements for the test suite.

- [ ] Run tests in parallel where possible. (ava is very good for this.)
- [ ] Move test suite out of `testutils.spec.ts`

**Comment by ethanfrey** [comment]
> Move test suite out of testutils.spec.ts

I would simply remove it. These helpers are stable and now covered indirectly from everything else. It was there to bootstrap

**Comment by ethanfrey** [comment]
> Run tests in parallel where possible. 

Yup, Most tests that don't do things like "queryAllClients()" should be runable in parallel. We can create 2 different connections with different ids at the same time.

It will also be a good test of the relayer in more difficult, real world situations.

What will likely break:
* Funding wallets (We submit multiple bank.send tx from the same account. It cannot sign more that one tx per block. We just need to make this controlled by a Mutex or something)
* Tests that use `allXyz()` queries - state changes outside of their control

I assume all tests make their own clients, so all connection/channel/packet tests are sandboxed. We could later test two relayers running in parallel on the same connection to ensure it is solid (different issue)
